### PR TITLE
Upgrade from AppWrapper v1beta1 to v1beta2

### PR DESCRIPTION
**Pull request by dgrove-oss**
# Issue link

# What changes have been made
Context: This PR supports the migration of the codeflare operator from MCADv1 to Kueue+AppWrapper.  We should probably wait on merging this PR until the corresponding PRs on the consuming codeflare repos are ready, as the removal of MCADv1 CRDs is a breaking change.

Change summary
1. Replace MCAD dependency with AppWrapper
2. Upgrade AppWrapper from v1beta1 to v1beta2
3. Upgrade to go 1.21
4. Upgrade go dependencies to match Kueue/AppWrapper requirements
5. Fix bug in fakeclient initialization (dynamic interface not initialized)

# Verification steps
I manually ran `go test ./support/. -v `

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ x ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Review by sutaakar — COMMENTED** [review]
/lgtm

**Review by Bobbins228 — COMMENTED** [review]
/lgtm

**Comment by dgrove-oss** [comment]
This PR is now ready for review.

**Review by KPostOffice — COMMENTED** [review]
Looks good! One coment

File: support/appwrapper.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/onsi/gomega"
+	mcadv1beta2 "github.com/project-codeflare/appwrapper/api/v1beta2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var appWrapperResource = mcadv1beta2.GroupVersion.WithResource("appwrappers")
+
+func AppWrapper(t Test, namespace string, name string) func(g gomega.Gomega) *mcadv1beta2.AppWrapper {
```
[the remark this answers is not in the document]
Comment (dgrove-oss) [reply support/appwrapper.go]: Change made.  This is an outlier (most similar functions in this package pass namespace as a string), but if this is the preferred new style I don't really care either way.

**Pull request merged**